config-service-impl/src/main/java/org/hypertrace/config/service/store/ConfigStore.java

```diff
@@ -42,6 +43,17 @@ UpsertedConfig writeConfig(
    */
   ContextSpecificConfig getConfig(ConfigResourceContext configResourceContext) throws IOException;
 
+  /**
+   * Get all the configs with the latest version(along with the context to which it applies) for the
+   * specified resource keys
+   *
+   * @param configResourceContexts
+   * @return the configs
+   * @throws IOException
+   */
+  Map<ConfigResourceContext, ContextSpecificConfig> getAllContextConfigs(
+      Set<ConfigResourceContext> configResourceContexts) throws IOException;
```

Comment:
Accept liberally. No reason that this needs to be a set, make it a collection (same on plural delete below).